app/services/rapid_ready_for_decision/processor_selector.rb

```diff
@@ -7,26 +7,38 @@ def initialize(form526_submission)
     end
 
     def processor_class
-      return nil unless rrd_enabled?
+      return unless rrd_enabled?
 
-      if rrd_enabled_disability?('hypertension') && single_issue_claim_applicable?(DiagnosticCodes::HYPERTENSION)
-        return RapidReadyForDecision::DisabilityCompensationJob
-      end
+      # single-issue claims only
+      return unless disability_types.size == 1
 
-      if rrd_enabled_disability?('asthma') && single_issue_claim_applicable?(DiagnosticCodes::ASTHMA)
-        return RapidReadyForDecision::Form526AsthmaJob
-      end
+      disability_type = disability_types.first
+      # must be a disability supported by RRD
+      return unless disability_type
+
+      # the RRD disability must be enabled
+      return unless rrd_enabled_disability?(disability_type)
 
-      nil
+      form_disability = form_disabilities.first
+      # the submitted disability must be for a claim for increase
+      return unless self.class.disability_increase?(form_disability, disability_type)
+
+      # TODO: use a constant hash instead
```

Comment:
Even better, why not just make this a value in each disability type? So, just

```ruby
return disability_type[:processor_class]
```

Maybe that's what you meant. Anyway, 👌  to do this later.

Reply:
I considered that, but I prefer not to put references to actual classes in the `Constants` -- only have primitive types -- to avoid circular dependencies. 
For the `TODO`, I meant something like the following defined in this class:
```ruby
DISABILITY_TO_PROCESSOR_CLASS = {
  RapidReadyForDecision::Constants::DISABILITIES[:hypertension] =>
    RapidReadyForDecision::DisabilityCompensationJob, 
  RapidReadyForDecision::Constants::DISABILITIES[:asthma] =>
    RapidReadyForDecision::Form526AsthmaJob
}.freeze
```
but it looks a bit messy.

Reply:
Yeah I did think about circular dependencies with the `Constants` route. Luckily, there's a way to do it with primitive types, so you get the best of both worlds:

```ruby
> 'RapidReadyForDecision::Form526AsthmaJob'.constantize
=> RapidReadyForDecision::Form526AsthmaJob
```

Reply:
Here's what I ended up doing: #9403